hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -740,22 +740,14 @@ private void clean(String startCleanTime) throws HoodieIOException  {
 
             List<String> partitionsToClean =
                 FSUtils.getAllPartitionPaths(fs, table.getMetaClient().getBasePath(), config.shouldAssumeDatePartitioning());
-            // shuffle to distribute cleaning work across partitions evenly
-            Collections.shuffle(partitionsToClean);
             logger.info("Partitions to clean up : " + partitionsToClean + ", with policy " + config
                 .getCleanerPolicy());
             if (partitionsToClean.isEmpty()) {
                 logger.info("Nothing to clean here mom. It is already clean");
                 return;
             }
 
-            int cleanerParallelism = Math.min(partitionsToClean.size(), config.getCleanerParallelism());
-            List<HoodieCleanStat> cleanStats = jsc.parallelize(partitionsToClean, cleanerParallelism)
-                    .map((Function<String, HoodieCleanStat>) partitionPathToClean -> {
-                        HoodieCleaner cleaner = new HoodieCleaner(table, config);
-                        return cleaner.clean(partitionPathToClean);
-                    })
-                .collect();
+            List<HoodieCleanStat> cleanStats = HoodieCleaner.clean(partitionsToClean, table, config, jsc);
```

Comment:
Sure Vinoth, I wasn't sure in which class I could put this method (since it requires a SparkContext), so made it static. I can change it the way compact is.